### PR TITLE
support perform multiple models delimited by comma

### PR DESCRIPTION
**Pull request by ZachCheung**
### perform all

```
$ ./gobackup perform
2022/12/07 00:18:49 [Config] Load config from default path.
2022/12/07 00:18:49 [Model: demo2] WorkDir: /tmp/gobackup3643219827/1670401129590681303/demo2
2022/12/07 00:18:49 [Database] => database | postgresql: postgresql
2022/12/07 00:18:49 [PostgreSQL] -> Dumping PostgreSQL...
2022/12/07 00:18:49 [PostgreSQL] dump path: /tmp/gobackup3643219827/1670401129590681303/demo2/postgresql/postgresql/demo.sql
2022/12/07 00:18:49 [Database] Dump succeeded
2022/12/07 00:18:49 [Compressor] => Compress | zst
2022/12/07 00:18:49 [Compressor] -> /tmp/gobackup3643219827/1670401129590681303/2022.12.07.00.18.49.tar.zst
2022/12/07 00:18:49 [Encryptor] => Encrypt | openssl
2022/12/07 00:18:49 [Encryptor] -> /tmp/gobackup3643219827/1670401129590681303/2022.12.07.00.18.49.tar.zst.enc
2022/12/07 00:18:49 [Splitter] Split to chunks
2022/12/07 00:18:49 [Splitter] Split done
2022/12/07 00:18:49 [Storage] => Storage | local
2022/12/07 00:18:49 [Local] Store succeeded /tmp/demo/2022.12.07.00.18.49
2022/12/07 00:18:49 [Model] Cleanup temp: /tmp/gobackup3643219827/
2022/12/07 00:18:49 [Model: demo] WorkDir: /tmp/gobackup3643219827/1670401129590745540/demo
2022/12/07 00:18:49 [Database] => database | postgresql: postgresql
2022/12/07 00:18:49 [PostgreSQL] -> Dumping PostgreSQL...
2022/12/07 00:18:49 [PostgreSQL] dump path: /tmp/gobackup3643219827/1670401129590745540/demo/postgresql/postgresql/demo.sql
2022/12/07 00:18:49 [Database] Dump succeeded
2022/12/07 00:18:49 [Compressor] => Compress | xz
2022/12/07 00:18:49 [Compressor] -> /tmp/gobackup3643219827/1670401129590745540/2022.12.07.00.18.49.tar.xz
2022/12/07 00:18:49 [Storage] => Storage | local
2022/12/07 00:18:49 [Local] Store succeeded /tmp/demo/2022.12.07.00.18.49.tar.xz
2022/12/07 00:18:49 [Model] Cleanup temp: /tmp/gobackup3643219827/
2022/12/07 00:18:49 [Model: demo1] WorkDir: /tmp/gobackup3643219827/1670401129590794856/demo1
2022/12/07 00:18:49 [Database] => database | postgresql: postgresql
2022/12/07 00:18:49 [PostgreSQL] -> Dumping PostgreSQL...
2022/12/07 00:18:50 [PostgreSQL] dump path: /tmp/gobackup3643219827/1670401129590794856/demo1/postgresql/postgresql/demo.sql
2022/12/07 00:18:50 [Database] Dump succeeded
2022/12/07 00:18:50 [Compressor] => Compress | zst
2022/12/07 00:18:50 [Compressor] -> /tmp/gobackup3643219827/1670401129590794856/2022.12.07.00.18.50.tar.zst
2022/12/07 00:18:50 [Encryptor] => Encrypt | openssl
2022/12/07 00:18:50 [Encryptor] -> /tmp/gobackup3643219827/1670401129590794856/2022.12.07.00.18.50.tar.zst.enc
2022/12/07 00:18:50 [Splitter] Split to chunks
2022/12/07 00:18:50 [Splitter] Split done
2022/12/07 00:18:50 [Storage] => Storage | local
2022/12/07 00:18:50 [Local] Store succeeded /tmp/demo/2022.12.07.00.18.50
2022/12/07 00:18:50 [Model] Cleanup temp: /tmp/gobackup3643219827/
```

### perform one

```
$ ./gobackup perform -m demo
2022/12/07 00:20:54 [Config] Load config from default path.
2022/12/07 00:20:54 [Model: demo] WorkDir: /tmp/gobackup3802922705/1670401254594830981/demo
2022/12/07 00:20:54 [Database] => database | postgresql: postgresql
2022/12/07 00:20:54 [PostgreSQL] -> Dumping PostgreSQL...
2022/12/07 00:20:54 [PostgreSQL] dump path: /tmp/gobackup3802922705/1670401254594830981/demo/postgresql/postgresql/demo.sql
2022/12/07 00:20:54 [Database] Dump succeeded
2022/12/07 00:20:54 [Compressor] => Compress | xz
2022/12/07 00:20:54 [Compressor] -> /tmp/gobackup3802922705/1670401254594830981/2022.12.07.00.20.54.tar.xz
2022/12/07 00:20:54 [Storage] => Storage | local
2022/12/07 00:20:54 [Local] Store succeeded /tmp/demo/2022.12.07.00.20.54.tar.xz
2022/12/07 00:20:54 [Cycler] Removed 2022.12.07.00.18.49.tar.xz
2022/12/07 00:20:54 [Model] Cleanup temp: /tmp/gobackup3802922705/
```

### perform multiple models

```
$ ./gobackup perform -m demo,demo2
2022/12/07 00:21:02 [Config] Load config from default path.
2022/12/07 00:21:02 [Model: demo] WorkDir: /tmp/gobackup3357776329/1670401262309155895/demo
2022/12/07 00:21:02 [Database] => database | postgresql: postgresql
2022/12/07 00:21:02 [PostgreSQL] -> Dumping PostgreSQL...
2022/12/07 00:21:02 [PostgreSQL] dump path: /tmp/gobackup3357776329/1670401262309155895/demo/postgresql/postgresql/demo.sql
2022/12/07 00:21:02 [Database] Dump succeeded
2022/12/07 00:21:02 [Compressor] => Compress | xz
2022/12/07 00:21:02 [Compressor] -> /tmp/gobackup3357776329/1670401262309155895/2022.12.07.00.21.02.tar.xz
2022/12/07 00:21:02 [Storage] => Storage | local
2022/12/07 00:21:02 [Local] Store succeeded /tmp/demo/2022.12.07.00.21.02.tar.xz
2022/12/07 00:21:02 [Cycler] Removed 2022.12.07.00.20.54.tar.xz
2022/12/07 00:21:02 [Model] Cleanup temp: /tmp/gobackup3357776329/
2022/12/07 00:21:02 [Model: demo2] WorkDir: /tmp/gobackup3357776329/1670401262309212975/demo2
2022/12/07 00:21:02 [Database] => database | postgresql: postgresql
2022/12/07 00:21:02 [PostgreSQL] -> Dumping PostgreSQL...
2022/12/07 00:21:02 [PostgreSQL] dump path: /tmp/gobackup3357776329/1670401262309212975/demo2/postgresql/postgresql/demo.sql
2022/12/07 00:21:02 [Database] Dump succeeded
2022/12/07 00:21:02 [Compressor] => Compress | zst
2022/12/07 00:21:02 [Compressor] -> /tmp/gobackup3357776329/1670401262309212975/2022.12.07.00.21.02.tar.zst
2022/12/07 00:21:02 [Encryptor] => Encrypt | openssl
2022/12/07 00:21:02 [Encryptor] -> /tmp/gobackup3357776329/1670401262309212975/2022.12.07.00.21.02.tar.zst.enc
2022/12/07 00:21:02 [Splitter] Split to chunks
2022/12/07 00:21:02 [Splitter] Split done
2022/12/07 00:21:02 [Storage] => Storage | local
2022/12/07 00:21:02 [Local] Store succeeded /tmp/demo/2022.12.07.00.21.02
2022/12/07 00:21:02 [Cycler] Removed 2022.12.07.00.18.49/2022.12.07.00.18.49.tar.zst.enc-000
2022/12/07 00:21:02 [Cycler] Removed 2022.12.07.00.18.49/2022.12.07.00.18.49.tar.zst.enc-001
2022/12/07 00:21:02 [Cycler] Removed 2022.12.07.00.18.49/
2022/12/07 00:21:02 [Model] Cleanup temp: /tmp/gobackup3357776329/
```

### CODE REVIEW

**Comment by huacnlee** [comment]
Try to implementation as `gobackup perform demo demo2`

```
Usage: gobackup perform [options] [model]...
```

Same behavior as `cargo install`

**Comment by ZachCheung** [comment]
No more `perform -m demo` for backward compatible?

**Comment by ZachCheung** [comment]
Or you mean `gobackup perform -m demo demo2`?

**Comment by ZachCheung** [comment]
```
$ ./gobackup perform -m demo
2022/12/07 05:24:27 [Config] Load config from default path.
2022/12/07 05:24:27 []string{"demo"}
```

```
$ ./gobackup perform -m demo demo2
2022/12/07 05:24:36 [Config] Load config from default path.
2022/12/07 05:24:36 []string{"demo", "demo2"}
```

```
$ ./gobackup perform demo demo2
2022/12/07 05:24:41 [Config] Load config from default path.
2022/12/07 05:24:41 []string{"demo", "demo2"}
```

```
$ ./gobackup perform -c /tmp/gobackup.yml demo demo2
2022/12/07 05:24:48 [Config] Load config: /tmp/gobackup.yml
2022/12/07 05:24:48 []string{"demo", "demo2"}
```

Backward compatible is dirty work.

File: main.go [hunk 1]
```diff
@@ -145,27 +141,24 @@ func main() {
...
-	modelConfig := config.GetModelByName(modelName)
-	if modelConfig == nil {
-		return
-	}
-	logger.Fatalf("Model %s not found in %s", modelName, viper.ConfigFileUsed())
 
-	m := model.Model{
-		Config: *modelConfig,
+	for _, m := range models {
+		m.Perform()
```
Comment (huacnlee) [inline main.go:162]: TODO: Perform model backups in parallel

Comment (ZachCheung) [reply main.go:162]: Then logger should have model tag.

**Pull request merged**